unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenPercentageController.cs

```diff
@@ -0,0 +1,79 @@
+using DCL.Controllers;
+using System;
+using UnityEngine;
+
+namespace DCL.LoadingScreen
+{
+    /// <summary>
+    /// Loading screen percentage updater. The responsibility of this class is to update the loading bar only
+    /// listening to the destination scene
+    /// </summary>
+    public class LoadingScreenPercentageController : IDisposable
+    {
+
+        private readonly LoadingScreenPercentageView loadingScreenPercentageView;
+        private readonly ISceneController sceneController;
+
+        private Vector2Int currentDestination;
+        private ParcelScene currentScene;
+
+
+        public LoadingScreenPercentageController(ISceneController sceneController, LoadingScreenPercentageView loadingScreenPercentageView)
+        {
+            this.loadingScreenPercentageView = loadingScreenPercentageView;
+            this.sceneController = sceneController;
+
+            loadingScreenPercentageView.gameObject.SetActive(false);
+
+            sceneController.OnNewSceneAdded += SceneController_OnNewSceneAdded;
+        }
+
+        public void Dispose()
+        {
+            sceneController.OnNewSceneAdded -= SceneController_OnNewSceneAdded;
+            if (currentScene)
+            {
+                currentScene.sceneLifecycleHandler.sceneResourcesLoadTracker.OnResourcesStatusUpdate -= ResourcesStatusUpdate;
+                currentScene.sceneLifecycleHandler.OnStateRefreshed -= Scene_OnStateRefreshed;
+            }
+        }
+
+        private void SceneController_OnNewSceneAdded(IParcelScene scene)
+        {
+            currentScene = scene as ParcelScene;
+            //We will only update the percentage of the current destination scene. It may be the only one we care about
+            if (currentScene != null &&
+                Environment.i.world.state.GetSceneNumberByCoords(currentDestination).Equals(currentScene.sceneData.sceneNumber))
+            {
+                loadingScreenPercentageView.SetLoadingPercentage(0);
+                currentScene.sceneLifecycleHandler.sceneResourcesLoadTracker.OnResourcesStatusUpdate += ResourcesStatusUpdate;
+                currentScene.sceneLifecycleHandler.OnStateRefreshed += Scene_OnStateRefreshed;
+            }
+        }
+
+        private void Scene_OnStateRefreshed(ParcelScene scene)
+        {
+            switch (scene.sceneLifecycleHandler.state)
+            {
+                case SceneLifecycleHandler.State.READY:
+                    currentScene.sceneLifecycleHandler.sceneResourcesLoadTracker.OnResourcesStatusUpdate -= ResourcesStatusUpdate;
+                    currentScene.sceneLifecycleHandler.OnStateRefreshed -= Scene_OnStateRefreshed;
+                    currentScene = null;
+                    break;
+            }
+        }
+
+        private void ResourcesStatusUpdate()
+        {
+            loadingScreenPercentageView.SetLoadingPercentage((int)currentScene.sceneLifecycleHandler.sceneResourcesLoadTracker.loadingProgress);
```

Comment:
I wonder what happens if the scene fails to load; should not it be somehow handled?

Reply:
Using the scene `74,47` for testing a failed scene, I discovered that there is no scene fail handling at all. 

In [kernel](https://github.com/decentraland/kernel/blob/bdbc933ea49af847ef3adac1ec17c30e25b26adf/packages/shared/world/SceneWorker.ts#L339) there is a SCENE_FAIL event called on timeout. But besides reporting that a scene failed loading, it does nothing.

I dont think that scene failing handling should be relative to this PR or even this epic at all. We should create a whole new ticket for SCENE_FAIL handling, since currently no backup system is set up.

WDYT?